internal/app/telescopes/api/handlers.go

```diff
@@ -70,7 +72,12 @@ func (r *RouteHandler) recommendClusterSetup() gin.HandlerFunc {
 			return
 		}
 
-		if response, err := r.engine.RecommendCluster(pathParams.Provider, pathParams.Service, pathParams.Region, req, nil, logger); err != nil {
+		vmSelector := vms.NewVmSelector(logger)
+		nodePoolSelector := nodepools.NewNodePoolSelector(logger)
+
+		engine := recommender.NewEngine(logger, vmSelector, nodePoolSelector, r.ciCli)
```

Comment:
what's the reasoning behind moving the creation of the engine component from the main package into handlers. I think the engine is the top component of this application and should / can be freely shared by handlers; there's no need to repeat the construction code whenever a request is received